graphql_compiler/tests/snapshot_tests/test_orientdb_match_query.py

```diff
@@ -45,10 +46,20 @@ def execute_graphql(
     if test_data.type_equivalence_hints:
         # For test convenience, we accept the type equivalence hints in string form.
         # Here, we convert them to the required GraphQL types.
-        schema_based_type_equivalence_hints = {
-            schema.get_type(key): schema.get_type(value)
-            for key, value in six.iteritems(test_data.type_equivalence_hints)
-        }
+        schema_based_type_equivalence_hints = {}
+        #     schema.get_type(key): schema.get_type(value)
+        #     for key, value in six.iteritems(test_data.type_equivalence_hints)
+        # }
+        for key, value in six.iteritems(test_data.type_equivalence_hints):
+            key_type = schema.get_type(key)
+            value_type = schema.get_type(value)
+            if (
+                key_type and value_type
+                and (isinstance(key_type, GraphQLInterfaceType)
+                     or isinstance(key_type, GraphQLObjectType))
+                and isinstance(value_type, GraphQLUnionType)
```

Comment:
Neat! You should be able to assert that this branch is always taken — the `else` here should `raise AssertionError`.